pkg/webreg/webreg.go

```diff
@@ -639,21 +711,142 @@ func jobHandler(regAgent load.RegistryAgent, confAgent load.ConfigAgent, w http.
 	writePage(w, "Job Test Workflow Help Page", page, workflow)
 }
 
+// addJob adds a test to the specified org, repo, and branch in the Jobs struct in alphabetical order
+func (j *Jobs) addJob(orgName, repoName, branchName, testName string) {
+	orgIndex := 0
+	orgExists := false
+	for _, currOrg := range j.Orgs {
+		if diff := strings.Compare(currOrg.Name, orgName); diff == 0 {
+			orgExists = true
+			break
+		} else if diff > 0 {
+			break
+		}
+		orgIndex++
+	}
+	if !orgExists {
+		newOrg := Org{Name: orgName}
+		j.Orgs = append(j.Orgs[:orgIndex], append([]Org{newOrg}, j.Orgs[orgIndex:]...)...)
+	}
+	repoIndex := 0
+	repoExists := false
+	for _, currRepo := range j.Orgs[orgIndex].Repos {
+		if diff := strings.Compare(currRepo.Name, repoName); diff == 0 {
+			repoExists = true
+			break
+		} else if diff > 0 {
+			break
+		}
+		repoIndex++
+	}
+	if !repoExists {
+		newRepo := Repo{Name: repoName}
+		repos := j.Orgs[orgIndex].Repos
+		j.Orgs[orgIndex].Repos = append(repos[:repoIndex], append([]Repo{newRepo}, repos[repoIndex:]...)...)
+	}
+	branchIndex := 0
+	branchExists := false
+	for _, currBranch := range j.Orgs[orgIndex].Repos[repoIndex].Branches {
+		if diff := strings.Compare(currBranch.Name, branchName); diff == 0 {
+			branchExists = true
+			break
+		} else if diff > 0 {
+			break
+		}
+		branchIndex++
+	}
+	if !branchExists {
+		newBranch := Branch{Name: branchName}
+		branches := j.Orgs[orgIndex].Repos[repoIndex].Branches
+		j.Orgs[orgIndex].Repos[repoIndex].Branches = append(branches[:branchIndex], append([]Branch{newBranch}, branches[branchIndex:]...)...)
+	}
+	// a single test shouldn't be added multiple times, but that case should be handled correctly just in case
+	testIndex := 0
+	testExists := false
+	for _, currTestName := range j.Orgs[orgIndex].Repos[repoIndex].Branches[branchIndex].Tests {
+		if diff := strings.Compare(currTestName, testName); diff == 0 {
+			testExists = true
+			break
+		} else if diff > 0 {
+			break
+		}
+		testIndex++
+	}
+	if !testExists {
+		tests := j.Orgs[orgIndex].Repos[repoIndex].Branches[branchIndex].Tests
+		j.Orgs[orgIndex].Repos[repoIndex].Branches[branchIndex].Tests = append(tests[:testIndex], append([]string{testName}, tests[testIndex:]...)...)
+	}
+}
+
 // getAllMultiStageTests return a map that has the config name in org-repo-branch format as the key and the test names for multi stage jobs as the value
-func getAllMultiStageTests(confAgent load.ConfigAgent) map[string][]string {
-	testMap := make(map[string][]string)
+func getAllMultiStageTests(confAgent load.ConfigAgent, jobAgent *prowConfig.Agent) *Jobs {
+	jobs := &Jobs{}
 	configs := confAgent.GetAll()
-	for filename, config := range configs {
+	allRepos := jobAgent.Config().AllRepos
+	for filename, releaseConfig := range configs {
 		name := strings.TrimSuffix(filename, ".yaml")
-		for _, test := range config.Tests {
+		var org, repo, branch string
```

Comment:
Do we not already have this logic somewhere? I would be surprised.

Reply:
Not in code I've written, since I haven't used a slice of `org/repo` strings to identify the org/repo/branch of a config file before. Maybe somewhere else in the code. I'll see if I can find anything.